File: user-api/ts-sdk/src/sdk/models/operations/updateuserv2.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class UpdateUserV2PathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class UpdateUserV2Request extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: UpdateUserV2PathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.UserV2;
}


export class UpdateUserV2Response extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  userV2?: shared.UserV2;
}

File: user-api/go-sdk/pkg/models/operations/updateuserv2.go
package operations

import (
	"openapi/pkg/models/shared"
)

type UpdateUserV2PathParams struct {
	ID string `pathParam:"style=simple,explode=false,name=id"`
}

type UpdateUserV2Request struct {
	PathParams UpdateUserV2PathParams
	Request    *shared.UserV2 `request:"mediaType=application/json"`
}

type UpdateUserV2Response struct {
	ContentType string
	StatusCode  int64
	UserV2      *shared.UserV2
}
